libnavigation-base/src/main/java/com/mapbox/navigation/base/trip/model/roadobject/bridge/BridgeEntrance.kt

```diff
@@ -0,0 +1,63 @@
+package com.mapbox.navigation.base.trip.model.roadobject.bridge
+
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObject
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectGeometry
+import com.mapbox.navigation.base.trip.model.roadobject.RoadObjectType
+
+/**
+ * Road object type that provides information about bridges on the route.
+ *
+ * @see RoadObject
+ * @see RoadObjectType.BridgeEntrance
+ */
+class BridgeEntrance private constructor(
+    distanceFromStartOfRoute: Double?,
+    objectGeometry: RoadObjectGeometry
+) : RoadObject(
+    RoadObjectType.BridgeEntrance,
+    distanceFromStartOfRoute,
+    objectGeometry
+) {
+
+    /**
+     * Transform this object into a builder to mutate the values.
+     */
+    fun toBuilder(): Builder = Builder(
+        objectGeometry
+    )
+        .distanceFromStartOfRoute(distanceFromStartOfRoute)
+
+    /**
+     * Returns a string representation of the object.
+     */
+    override fun toString(): String {
+        return "BridgeEntrance() ${super.toString()}"
+    }
+
+    /**
+     * Use to create a new instance.
+     *
+     * @see BridgeEntrance
+     */
+    class Builder(
+        private val objectGeometry: RoadObjectGeometry
+    ) {
+        private var distanceFromStartOfRoute: Double? = null
+
+        /**
+         * Add optional distance from start of route.
+         */
+        fun distanceFromStartOfRoute(distanceFromStartOfRoute: Double?): Builder = apply {
```

Comment:
might that be 0 or less than 0? could you also add tests if it has some limitations

Reply:
We get it from `Directions`,  hard to say if they can send a negative value. 
I guess we can check it and ignore if it's negative.
cc @LukasPaczos @Guardiola31337

Reply:
I don't think this could ever be negative (distance from the start of the route to an intersection/any road object).

/cc @danpat @PureW

Reply:
This is the distance in the waypoint object in the route response?

> The straight-line distance from the coordinate specified in the query to the location it was snapped to.

I'd say that a well-behaved router should never return negative distances. But we have seen overflows in the past resulting in negative values for things that never _should_ be negative in route responses.

Reply:
> I'd say that a well-behaved router should never return negative distances. 

imo we should rely on it, but because it's not an outside developer responsibility let's replace negative by `0`(and avoid throws an exception)

Reply:
I guess `null` instead of negative is better than `0`. 
`null` says that we  have an `invalid` distance.
`0` can be used as a `valid` distance and may confuse cutomers

Reply:
make sense, I'm fine with both. Anyway, should mention it in kdocs

Reply:
@RingerJK docs added, tests added